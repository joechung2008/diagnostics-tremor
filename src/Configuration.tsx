import {
  TableRoot,
  Table,
  TableHead,
  TableHeaderCell,
  TableBody,
  TableRow,
  TableCell,
} from "@/components/Table";

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const items = Object.entries(config).reduce<KeyValuePair<string>[]>(
    (previous, [key, value]) => [...previous, { key, value }],
    []
  );

  return (
    <div>
      <h3 className="text-2xl">Configuration</h3>
      <TableRoot>
        <Table aria-label="Configuration">
          <TableHead>
            <TableRow>
              <TableHeaderCell>Key</TableHeaderCell>
              <TableHeaderCell>Value</TableHeaderCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {items.map((item, idx) => (
              <TableRow key={idx}>
                <TableCell>{item.key}</TableCell>
                <TableCell>{item.value}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableRoot>
    </div>
  );
};

export default Configuration;
