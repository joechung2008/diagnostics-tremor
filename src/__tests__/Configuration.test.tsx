import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Configuration from "../Configuration";

describe("Configuration", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Configuration config={{ key1: "value1", key2: "value2" }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
