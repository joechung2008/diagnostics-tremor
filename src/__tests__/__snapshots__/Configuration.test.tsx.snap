// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Configuration > should render correctly 1`] = `
<div>
  <h3
    class="text-2xl"
  >
    Configuration
  </h3>
  <div>
    <div
      class="w-full overflow-auto whitespace-nowrap"
    >
      <table
        aria-label="Configuration"
        class="w-full caption-bottom border-b border-gray-200 dark:border-gray-800"
        tremor-id="tremor-raw"
      >
        <thead
          class=""
        >
          <tr
            class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
          >
            <th
              class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
            >
              Key
            </th>
            <th
              class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          class="divide-y divide-gray-200 dark:divide-gray-800"
        >
          <tr
            class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
          >
            <td
              class="p-4 text-sm text-gray-600 dark:text-gray-400"
            >
              key1
            </td>
            <td
              class="p-4 text-sm text-gray-600 dark:text-gray-400"
            >
              value1
            </td>
          </tr>
          <tr
            class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
          >
            <td
              class="p-4 text-sm text-gray-600 dark:text-gray-400"
            >
              key2
            </td>
            <td
              class="p-4 text-sm text-gray-600 dark:text-gray-400"
            >
              value2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
